projects/demo/src/modules/components/pdf-viewer/examples/2/index.ts

```diff
@@ -0,0 +1,51 @@
+import {Component, Inject} from '@angular/core';
+import {TuiDialog} from '@taiga-ui/cdk';
+import {TuiNotificationsService} from '@taiga-ui/core';
+import {TuiPdfViewerOptions, TuiPdfViewerService} from '@taiga-ui/kit';
+import {PolymorpheusComponent} from '@tinkoff/ng-polymorpheus';
+import {switchMap} from 'rxjs/operators';
+import {changeDetection} from '../../../../../change-detection-strategy';
+import {encapsulation} from '../../../../../view-encapsulation';
+import {ActionsContent} from './actions-content.component';
+import {PdfContent} from './pdf-content.component';
+
+export type Buttons = readonly Readonly<{
+    text: string;
+    onClick(context: TuiDialog<TuiPdfViewerOptions<Buttons>, string>): void;
+}>[];
+
+@Component({
+    selector: 'tui-pdf-viewer-example-2',
+    templateUrl: './index.html',
+    changeDetection,
+    encapsulation,
+})
+export class TuiPdfViewerExample2 {
+    constructor(
+        @Inject(TuiNotificationsService)
+        private readonly notifications: TuiNotificationsService,
+        @Inject(TuiPdfViewerService) private readonly pdfService: TuiPdfViewerService,
+    ) {}
+
+    show() {
+        const options: TuiPdfViewerOptions<Buttons> = {
+            label: 'Taiga UI',
+            actions: new PolymorpheusComponent(ActionsContent),
+            data: [
+                {
+                    text: 'Sign',
+                    onClick: context => context.completeWith('Document signed'),
+                },
+                {
+                    text: 'Deny',
+                    onClick: context => context.completeWith('Document denied'),
+                },
+            ],
+        };
+
+        this.pdfService
+            .open<string>(new PolymorpheusComponent(PdfContent), options)
```

Comment:
why not encapsulate this behavior internally?

```ts
this.pdfService.open<string>(PdfContent, options)
```

for easy usage

Reply:
Because it works with PolymorpheusContent like everything else.